src/Postmark/PostmarkClient.cs

```diff
@@ -953,11 +968,14 @@ public async Task<PostmarkResponse> SendEmailWithTemplateAsync(TemplatedPostmark
             IDictionary<string, string> metadata = null,
             params PostmarkMessageAttachment[] attachments)
         {
-            
+
             var email = new TemplatedPostmarkMessage();
-            if(templateReference is long){
+            if (templateReference is long)
+            {
                 email.TemplateId = (long)templateReference;
```

Comment:
Should the `TemplateId` be nullable? Checking the backend sending route we check if template alias/id are both null. Here if templateReference is not a `long`, TempalteId will be default(long) of 0. https://wb.beanstalkapp.com/postmark-delivery/browse/git/src/Web/Controllers/EmailController.cs#L1677743731

Along the same lines but not specific to this review, it seems on the `EmailSendRequest` in `postmark-delivery` we should be checking both TemplateId and TemplateAlias for value on `IsTemplatedSendRequest`. I must have missed that in the larger template review. https://wb.beanstalkapp.com/postmark-delivery/browse/git/src/Core/MailSend/EmailSendRequest.cs#L3864477103